### ISSUE TITLE
Upgrade from Kubernetes 1.5.6 to 1.6.x breaks weave

### ISSUE THREAD

**Issue by caarlos0** [kubernetes/kops#2366]
Seems like weave daemonset was not upgraded when I upgraded the cluster.

As a workaround I manually deleted the daemon set and re-created it following weave's docs:

```
kubectl delete daemonset weave-net -n kube-system
kubectl apply -f https://git.io/weave-kube-1.6
```

Tested with kops 1.6.0-alpha.2

**Comment by cleberwarmling** [comment]
Se pa. Já até corrigiram https://github.com/kubernetes/kops/pull/2353

**Comment by keiths-osc** [comment]
I've noticed the same thing and had logged a [separate issue](https://github.com/kubernetes/kops/issues/2377) with some additional details about what I've found with weave so far.  (At first I didn't realize my entire problem was that the weave upgrade wasn't happening, or I would have added details to this issue instead).

**Comment by caarlos0** [comment]
Yeah, seems like basically it never upgraded weave... upgrading manually works though.

**Comment by hollowimage** [comment]
From what you guys have done, does it make sense to upgrade waeve first, and then upgrade the cluster?

**Comment by caarlos0** [comment]
@hollowimage probably will not work. As far as I know weave for k8s1.6 has some things different that doesn't work on k8s1.5...

**Comment by hollowimage** [comment]
Got it. so essentially until this is fixed, during the upgrade process the entire cluster becomes vulnerable and essentially down until weave is reintroduced?

**Comment by caarlos0** [comment]
At least in my case, that was what happened.

**Comment by chrislovecnm** [comment]
@hollowimage you can update the YAML, and as each node is replaced the new version of weave should start.  You need to do a replace, not a delete.  The YAML for weave 1.6 is different.  Personally, I would file an issue in the weave repo, and see if they can provide documentation.

If you are using RBAC with 1.6, stuff with weave changes.  Otherwise there 'should' not be compatibility issues.

1. test this process, as I have not :)
1. kubectl upgrade the ds
1. rolling update nodes in order to upgrade
1. new nodes should fire up new pods with new version of weave

**Comment by hollowimage** [comment]
Ive done weave upgrades before and they usually require a force-pod kill one by one after updating the ds. that being said the issue here is that ideally both weave and k8s upgrade happen at the same time, so at no point is the cluster stuck in a bad state,

either way, thanks. This confirms what i was thinking on doing either way.

**Comment by chrislovecnm** [comment]
@hollowimage so the force pod kill was because of a bug in weave :)  Are you still getting this in 1.9?

**Comment by hollowimage** [comment]
@chrislovecnm did not realize it was fixed. I did it anyway because it was a dev cluster and we only had 2 nodes there. But I am slating the upgrade of our production cluster soon (6 minions, 3 masters) and we will see how that works.

**Comment by chrislovecnm** [comment]
The bug about the force kill should have been fixed in 1.8, I have not tested weave on k8s 1.6, but I have heard that it is working.

**Comment by arthurlm44** [comment]
I had to do this manually as well.

**Comment by jordanjennings** [comment]
In my testing I saw that the ManagedFile for weave was not updated in the state store when doing an upgrade. I created a cluster with kops 1.5 and k8s 1.5.3, then did an upgrade with kops 1.6 and k8s 1.5.5, but the new yaml wasn't put into the state store and the DaemonSet wasn't updated. The v1.9.3.yaml file was still in the state store and the new k8s-1.6.yaml and pre-k8s-1.6.yaml files were not there.
